packages/contracts/src/deploy-utils.ts

```diff
@@ -97,7 +70,22 @@ export const deploy = async (
     ),
   })
 
-  if (result.newlyDeployed && typeof opts.postDeployAction === 'function') {
+  if (result.newlyDeployed) {
+    // Always wait for the transaction to be mined, just in case.
+    await hre.ethers.provider.waitForTransaction(result.transactionHash)
+    console.log(`deployed ${name} at ${result.address}`)
+  } else {
+    // If the contract is not newly deployed, do not proceed further.
+    return created
+  }
```

Comment:
```suggestion
  // If the contract is not newly deployed, do not proceed further.
  if (!result.newlyDeployed) {
    return created
  }

  // Always wait for the transaction to be mined, just in case.
  await hre.ethers.provider.waitForTransaction(result.transactionHash)
  console.log(`deployed ${name} at ${result.address}`)
```